src/classes/QueryBuilder.cls

```diff
@@ -2,41 +2,54 @@
 * This file is part of the Nebula Framework project, released under the MIT License.             *
 * See LICENSE file or go to https://github.com/jongpie/NebulaFramework for full license details. *
 *************************************************************************************************/
-public without sharing class QueryBuilder extends NebulaCore implements IQueryBuilder {
+public without sharing virtual class QueryBuilder extends NebulaCore implements IQueryBuilder {
```

Comment:
Do you have specific plans for extending the class in your implementation? Or do you think we should make all the main classes in the framework be virtual?